File: lihaiyang/cpp/DestinationCity.cpp
// DestinationCity.cpp
// Author: hyan23
// 2020.05.31
// https://leetcode.com/problems/destination-city/

// Runtime: 32 ms, faster than 45.94% of C++ online submissions for Destination City.
// Memory Usage: 12.1 MB, less than 100.00 % of C++ online submissions for Destination City.

#include <iostream>
#include <vector>
#include <set>
#include <algorithm>

using namespace std;

class Solution {
public:
	string destCity(vector<vector<string>>& paths) {
		set<string> s;
		for_each(paths.cbegin(), paths.cend(), [&](auto e) { s.insert(e[1]); });
		for_each(paths.cbegin(), paths.cend(), [&](auto e) { s.erase(e[0]); });
		return *s.begin();
	}
};

int main(void)
{
	return 0;
}

File: huangguoyu/week_4_of_May/destination-city.java
class Solution {
    public String destCity(List<List<String>> paths) {
        Map<String, String> routes = new HashMap<>();
        for(List<String> item : paths) {
            routes.put(item.get(0), item.get(1));
        }

        for(List<String> item : paths) {
            if (routes.get(item.get(0)) == null) 
                return item.get(0);
            if (routes.get(item.get(1)) == null) 
                return item.get(1);
        }
        return null;
    }
}